### PR TITLE
Feature: Get instance

### LINKED ISSUE #46: Supporting getting instance by name
When multiple services interact with the same backend, it's useful to easily pass instance by name to interact with it.

### PR DESCRIPTION
**Pull request by plule-ansys**
Add the feature to get an instance by name.

Fixes #46

### CODE REVIEW

File: src/ansys/platform/instancemanagement/client.py [hunk 1]
```diff
@@ -263,3 +265,36 @@ def create_instance(
...
+            Name of the instance to get (for example ``instances/mapdl-1212``).
+
+        Returns
+        -------
+        Instance
+            A remote instance.
+
+        timeout : float, optional
+            Maximum time in seconds for the request. The default is ``None``.
+
```
Comment (jerome-blanche) [inline src/ansys/platform/instancemanagement/client.py:284]: doc for the timeout parameter to move in the Parameters section?

File: tests/test_client.py [hunk 2]
```diff
@@ -150,6 +150,92 @@ def server():
...
+                },
+            )
+        )
+        rpc.terminate(instance, [], StatusCode.OK, "")
+        return request
+
+    server_future = testing_pool.submit(server)
+
+    # Act
+    # Getthe instance from the client
```
Comment (jerome-blanche) [inline tests/test_client.py:183]: typo: Get the

File: tests/test_client.py [hunk 3]
```diff
@@ -150,6 +150,92 @@ def server():
...
+        },
+    )
+
+
+def test_update_notfound(
+    testing_pool: ThreadPoolExecutor,
+    testing_channel: grpc_testing.Channel,
+):
+    # Arrange
+    # A failing to provide instances
```
Comment (jerome-blanche) [inline tests/test_client.py:209]: missing a word? A server failing

**Review by jerome-blanche — CHANGES_REQUESTED** [review]

File: src/ansys/platform/instancemanagement/client.py (hunk 1 above)
Comment (plule-ansys) [reply src/ansys/platform/instancemanagement/client.py]: Yes! thank you

**Review by jerome-blanche — APPROVED** [review]

File: src/ansys/platform/instancemanagement/client.py [hunk 4]
```diff
@@ -263,3 +265,36 @@ def create_instance(
...
         definition = definitions[0]
         return definition.create_instance(timeout=requests_timeout)
+
+    def get_instance(self, name: str, timeout: float = None) -> Instance:
+        """Get a remote product instance by name.
+
+        Parameters
+        ----------
+        name: str
+            Name of the instance to get (for example ``instances/mapdl-1212``).
```
Comment (PipKat) [inline src/ansys/platform/instancemanagement/client.py:275]:
```suggestion
            Name of the instance to get (for example, ``instances/mapdl-1212``).
```

File: src/ansys/platform/instancemanagement/client.py [hunk 5]
```diff
@@ -263,3 +265,36 @@ def create_instance(
...
+
+        timeout : float, optional
+            Maximum time in seconds for the request. The default is ``None``.
+
```
Comment (PipKat) [inline src/ansys/platform/instancemanagement/client.py:279]: The default is ``None``, in which case ? where ? is something like "there is no maximum time out". What happens when the default is used?

**Review by PipKat — APPROVED** [review]

File: src/ansys/platform/instancemanagement/client.py (hunk 5 above)
Comment (PipKat) [reply src/ansys/platform/instancemanagement/client.py:279]: The above comment is true for all defaults where the behavior is not obvious. This is almost always true when the default is ``None``.`

Comment (plule-ansys) [reply src/ansys/platform/instancemanagement/client.py:279]: Yes, I agree, I've created #50 to track this

**Pull request merged**